core/src/main/java/io/questdb/cairo/sql/async/PageFrameReduceJob.java

```diff
@@ -188,8 +189,9 @@ private static boolean consumeQueue(
                         reduce(workerId, record, circuitBreaker, task, frameSequence, stealingFrameSequence);
                     }
                 } catch (Throwable e) {
+                    LOG.error().$("reduce error [ex=").$(e).I$();
+                    task.setErrorMsg(e.getMessage() != null ? e.getMessage() : exceptionMessage);
```

Comment:
this is tripple string creation in case of an error. There is a `FlyweightMessageContainer` interface that most our exception systems implement. This provides error text without incurring string creation.

Calling getMessage() will result in `String` object created once per call.

This code can be rewritten such that `task` has `StringSink` where error message contents can be copied to. This in effect the same copying String is doing without extra object allocation.

Reply:
Yes, I'm aware of the flyweight interface. I wanted to keep object structure plain, but considering recent changes in the `StringSink` class it makes no sense.

Addressed in 5158efa5bf0adb2deb484d01d5cb2a2bb2d2af44